std/functional.d

```diff
@@ -260,6 +210,136 @@ unittest
     static assert(!is(typeof(binaryFun!FuncObj)));
 }
 
+/**
+Transforms a string representing an expression into a n-ary function.
+$(UL
+    $(LI The number of parameters is given by $(D argc). If no explicit
+        $(D argc) is given, it is inferred to be $(D parmNames.length).)
+    $(LI Explicit names for the function parameters may instead be given as
+        $(D parmNames). If more than $(D argc) names are supplied, the
+        extras will be ignored.)
+    $(LI If no $(D parmNames) are given, the letters ($(D a) through $(D z))
+        are used to name up to the first 26 function parameters.)
+    $(LI $(D args[index]) can always be used to access any parameter by index,
+        regardless of whether it has been given a name.)
+)
+*/
+template naryFun(alias fun, size_t argc, parmNames...)
+    if (parmNames.length == 0 || allSatisfy!(isSomeString, typeof(parmNames)))
+{
+    static if (is(typeof(fun) : string))
+    {
+        // No need to regenerate the aliases mixin for every ElementTypes permutation.
+        private enum aliases = naryFunAliases(argc, parmNames);
+
+        static if (!fun._ctfeMatchNary(parmNames))
+        {
+            import std.traits, std.typecons, std.typetuple;
+            import std.algorithm, std.conv, std.exception, std.math, std.range, std.string;
+        }
+
+        auto naryFun(ElementTypes...)(auto ref ElementTypes args)
+            if (ElementTypes.length == argc)
+        {
+            mixin(aliases);
+            return mixin(fun);
+        }
+    }
+    else static if (needOpCallAlias!fun)
+    {
+        // Issue 9906
+        alias naryFun = fun.opCall;
+    }
+    else
+    {
+        alias naryFun = fun;
+    }
+}
+
+/// ditto
+template naryFun(alias fun, parmNames...)
+    if (allSatisfy!(isSomeString, typeof(parmNames)))
+{
+    alias naryFun = naryFun!(fun, parmNames.length, parmNames);
+}
+
+// Generate naryFun parameter aliases mixin string.
+string naryFunAliases(size_t argc, string[] parmNames...) pure
```

Comment:
I somehow dropped the `private` access modifier here, by accident. Please put it back.